### PR TITLE
feat: Add the ability to delete/archive articles

### PR DESCRIPTION
**Pull request by muhsin-k**
Fixes https://github.com/chatwoot/product/issues/534

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/javascript/dashboard/routes/dashboard/helpcenter/pages/articles/EditArticle.vue [hunk 1]
```diff
@@ -27,6 +27,17 @@
...
+      @archive-article="archiveArticle"
+    />
+    <woot-delete-modal
+      :show.sync="enableDeleteConfirmationPopup"
+      :on-close="closeDeletePopup"
+      :on-confirm="confirmDeletion"
+      :title="$t('HELP_CENTER.DELETE_ARTICLE.MODAL.CONFIRM.TITLE')"
+      :message="$t('HELP_CENTER.DELETE_ARTICLE.MODAL.CONFIRM.MESSAGE')"
+      :confirm-text="$t('HELP_CENTER.DELETE_ARTICLE.MODAL.CONFIRM.YES')"
+      :reject-text="$t('HELP_CENTER.DELETE_ARTICLE.MODAL.CONFIRM.YES')"
```
Comment (iamsivin) [inline app/javascript/dashboard/routes/dashboard/helpcenter/pages/articles/EditArticle.vue:40]:
```suggestion
      :reject-text="$t('HELP_CENTER.DELETE_ARTICLE.MODAL.CONFIRM.NO')"
```

File: app/javascript/dashboard/routes/dashboard/helpcenter/pages/articles/EditArticle.vue [hunk 2]
```diff
@@ -52,6 +63,7 @@ export default {
       isSaved: false,
       showArticleSettings: false,
       alertMessage: '',
+      enableDeleteConfirmationPopup: false,
```
Comment (iamsivin) [inline app/javascript/dashboard/routes/dashboard/helpcenter/pages/articles/EditArticle.vue:66]:
```suggestion
      showDeleteConfirmationPopup: false,
```

File: app/javascript/dashboard/routes/dashboard/helpcenter/pages/articles/EditArticle.vue [hunk 3]
```diff
@@ -83,6 +95,16 @@ export default {
         portalSlug: this.selectedPortalSlug,
       });
     },
+    openDeletePopup() {
+      this.enableDeleteConfirmationPopup = true;
```
Comment (iamsivin) [inline app/javascript/dashboard/routes/dashboard/helpcenter/pages/articles/EditArticle.vue:99]:
```suggestion
      this. showDeleteConfirmationPopup = true;
```

**Review by iamsivin — COMMENTED** [review]
@muhsin-k Please see the comments added.

**Review by iamsivin — COMMENTED** [review]
@muhsin-k Tested locally and found some console error while deleting the article.
https://www.loom.com/share/90a891fe55da43fc87906f8d7e07ba72

File: app/javascript/dashboard/routes/dashboard/helpcenter/components/PortalPopover.vue [hunk 4]
```diff
@@ -61,10 +61,16 @@ export default {
...
       this.$emit('close-popover');
     },
-    openPortalPage() {
+    openPortalPage({ slug, locale }) {
       this.$emit('close-popover');
+      const portal = this.portals.find(p => p.slug === slug);
+      this.$store.dispatch('portals/setPortalId', portal.id);
       this.$router.push({
-        name: 'list_all_portals',
+        name: 'list_all_locale_articles',
```
Comment (nithindavid) [inline app/javascript/dashboard/routes/dashboard/helpcenter/components/PortalPopover.vue:69]: @muhsin-k The function name `openPortalPage` and route `list_all_locale_articles` doesn't seem related, Should we have better name?

**Review by nithindavid — DISMISSED** [review]

**Review by iamsivin — APPROVED** [review]
LGTM
